client/src/components/modal-forms/AssignCohort.js

```diff
@@ -0,0 +1,143 @@
+import { Button } from '../generic';
+import { Box, Typography } from '@material-ui/core';
+import { RenderSelectField } from '../fields';
+import { Field, Formik, Form as FormikForm } from 'formik';
+import { ParticipantAssignCohortSchema } from '../../constants';
+import { psi, assignParticipantWithCohort } from '../../services';
+
+// Helpers
+const fetchPSIData = async () => {
+  const psiData = (await psi()) || [];
+  const cohortMap = {};
+  const psiMappedData = psiData.map((psi) => {
+    const mappedData = {
+      value: psi.id,
+      label: psi.institute_name,
+    };
+    const cohorts = psi.cohorts.map((cohort) => ({
+      value: cohort.id,
+      label: cohort.cohort_name,
+    }));
+    cohortMap[psi.id] = cohorts;
+    return mappedData;
+  });
+
+  return { psiMappedData, cohortMap, psiData };
+};
+
+// Post Cohort Assignment Form
+const postCohortAssignmentForm = async ({
+  values,
+  participantId,
+  afterPostAction,
+  setShowError,
+  cachedPSIData,
+}) => {
+  try {
+    await assignParticipantWithCohort({ participantId, cohortId: values.cohort });
+    const psi = cachedPSIData.find((psi) => psi.id === values.institute);
+    const cohort = psi.cohorts.find((cohort) => cohort.id === values.cohort);
+    afterPostAction(cohort);
+  } catch (error) {
+    setShowError(true);
+  }
+};
+
+export const AssignCohortForm = ({ initialValues = {}, onClose, onSubmit, participantId }) => {
+  // State: PSI List to fill the select field
+  const [psiList, setPSIList] = useState([]);
+  // State: Cohort list connected with the PSI
+  const [cohorts, setCohorts] = useState([]);
+  // State Cohort Map
+  const [cohortMapData, setCohortMapData] = useState({});
+  // State: Cached PSI data
+  const [cachedPSIData, setCachedPSIData] = useState([]);
+
+  // Show Error
+  const [showError, setShowError] = useState(false);
+
+  // UseEffect: Fetch PSI List
+  useEffect(() => {
+    fetchPSIData().then(({ psiMappedData, cohortMap, psiData }) => {
+      setPSIList(psiMappedData);
+      setCohortMapData(cohortMap);
+      setCachedPSIData(psiData);
+    });
+  }, [setPSIList, setCohortMapData, setCachedPSIData]);
+
+  return (
+    <Box spacing={8} p={10}>
+      <Formik
+        initialValues={initialValues}
+        validationSchema={ParticipantAssignCohortSchema}
+        onSubmit={(values) =>
+          postCohortAssignmentForm({
+            values,
+            participantId,
+            afterPostAction: onSubmit,
+            setShowError,
+            cachedPSIData,
+          })
+        }
+      >
+        {({ submitForm, values, setFieldValue }) => (
+          <FormikForm>
+            <Typography color={'primary'} variant={'h4'}>
+              {' '}
+              Assign Cohort
+            </Typography>
+            <hr />
+            <MuiAlert severity='warning'>
+              {
+                'This Participant has not been assigned a cohort. Please assign a cohort to track graduation.'
```

Comment:
Doesn't match design copy

- participant should be lowercase
- "in order to track graduation" should be the ending